### PR TITLE
WRP-3063: Increased code coverage for Image, Input, Keypad and Tabgroup

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Increased code coverage for Image, Input, Keypad and Tabgroup

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRP-3063

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian [daniel.stoian@lgepartner.com](mailto:daniel.stoian@lgepartner.com)

### CODE REVIEW

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

**Review by alexandrumorariu — APPROVED** [review]
LGTM

File: Image/tests/Image-specs.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import '@testing-library/jest-dom';
+import {fireEvent, render, screen} from '@testing-library/react';
+
+import Image from '../Image';
```
Comment (jeonghee27) [inline Image/tests/Image-specs.js:4]: How about adding a tc that verifies the `sizing` prop just like sandstone?

File: Input/tests/Input-specs.js [hunk 2]
```diff
@@ -151,28 +193,23 @@ describe('Input Specs', () => {
 		const inputText = screen.getByLabelText('Input field').children[0];
 
 		userEvent.type(inputText, value);
+		// bluring input onSpotlightLeft for code coverage purposes
```
Comment (jeonghee27) [inline Input/tests/Input-specs.js:196]: onSpotlightRight?

File: TabGroup/tests/TabGroup-specs.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ import {render, screen} from '@testing-library/react';
```
Comment (jeonghee27) [inline TabGroup/tests/TabGroup-specs.js:3]: tabGroup's code coverage is already 100%, could you please add tests for onClick and icon (should render icon)? I wish you could get it from sandstone to save time.

File: Input/tests/Input-specs.js (hunk 2 above)
Comment (daniel-stoian-lgp) [reply Input/tests/Input-specs.js:196]: 👍

File: TabGroup/tests/TabGroup-specs.js (hunk 3 above)
Comment (daniel-stoian-lgp) [reply TabGroup/tests/TabGroup-specs.js:3]: I added a test for rendering icons, but the test for onClick does not trigger the event :(
The Tab component has an onClick prop, but it doesn't appear to be called in the test :(
Here is the test that I tried:

	test('should fire `onClick` with `onClick` type when a tab is clicked', () => {
		const handleClick = jest.fn();
		render(
			<TabGroup
				tabPosition="before"
				tabs={[
					{title: 'Home', icon: 'home', onClick: handleClick},
					{title: 'Settings', icon: 'setting'},
					{title: 'Theme', icon: 'display'}
				]}
			/>
		);

		userEvent.click(screen.getByRole('group').children[0]); //clicking tab

		expect(handleClick).toHaveBeenCalled();
	});

Comment (jeonghee27) [reply TabGroup/tests/TabGroup-specs.js:3]: It seems that onClick isn't supplied to \<Tab>.
I think you found a bug.
It's very strange.The onClick has not been used anywhere, and I don't know how the tabGroup is actually being used.

I think Tab click behavior should be fixed, and it seems necessary to modify the related sampler and add tests. Could you create a CLM issue and see it as a very low priority? @daniel-stoian-lgp

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: TabGroup/tests/TabGroup-specs.js (hunk 3 above)
Comment (daniel-stoian-lgp) [reply TabGroup/tests/TabGroup-specs.js:3]: Sure, we'll create a CLM issue. Already on our TODO list

**Pull request merged**